src/ast/query.rs

```diff
@@ -974,6 +974,8 @@ pub enum TableFactor {
         with_ordinality: bool,
         /// [Partition selection](https://dev.mysql.com/doc/refman/8.0/en/partitioning-selection.html), supported by MySQL.
         partitions: Vec<Ident>,
+        /// Optional PartiQL JsonPath: <https://partiql.org/dql/from.html>
+        partiql: Option<JsonPath>,
```

Comment:
Maybe we can call this `json_path` to be generic (in the doc we mention partiql as one example)?